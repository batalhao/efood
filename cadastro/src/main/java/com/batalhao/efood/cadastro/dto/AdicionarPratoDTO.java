package com.batalhao.efood.cadastro.dto;

import java.math.BigDecimal;
import com.batalhao.efood.cadastro.infra.DTO;
import com.batalhao.efood.cadastro.infra.ValidDTO;

@ValidDTO
public class AdicionarPratoDTO implements DTO {

  public String nome;

  public String descricao;

  public BigDecimal preco;

}
