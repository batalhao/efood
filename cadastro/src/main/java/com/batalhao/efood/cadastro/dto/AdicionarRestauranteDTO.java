package com.batalhao.efood.cadastro.dto;

public class AdicionarRestauranteDTO {

  public String proprietario;

  public String cnpj;

  public String nomeFantasia;

  public LocalizacaoDTO localizacao;

}
