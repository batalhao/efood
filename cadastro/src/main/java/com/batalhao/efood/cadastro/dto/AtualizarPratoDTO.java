package com.batalhao.efood.cadastro.dto;

import java.math.BigDecimal;

public class AtualizarPratoDTO {

  public BigDecimal preco;

}
