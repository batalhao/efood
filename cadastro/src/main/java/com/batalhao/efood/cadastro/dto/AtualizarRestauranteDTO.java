package com.batalhao.efood.cadastro.dto;

public class AtualizarRestauranteDTO {

  public String nomeFantasia;

}
