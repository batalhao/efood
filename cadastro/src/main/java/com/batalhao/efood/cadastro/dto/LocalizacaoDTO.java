package com.batalhao.efood.cadastro.dto;

public class LocalizacaoDTO {

  public Double latitude;

  public Double longitude;

}
