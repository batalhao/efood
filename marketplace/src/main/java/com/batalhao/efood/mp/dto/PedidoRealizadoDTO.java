package com.batalhao.efood.mp.dto;

import java.util.List;

public class PedidoRealizadoDTO {

  public List<PratoPedidoDTO> pratos;

  public RestauranteDTO restaurante;

  public String cliente;

}
