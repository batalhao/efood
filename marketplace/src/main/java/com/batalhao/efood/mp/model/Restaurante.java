package com.batalhao.efood.mp.model;

public class Restaurante {

  public Long id;

  public String nome;

  public Localizacao localizacao;

}
