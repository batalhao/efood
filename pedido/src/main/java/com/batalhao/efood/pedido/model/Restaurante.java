package com.batalhao.efood.pedido.model;

public class Restaurante {

  public String nome;

}
