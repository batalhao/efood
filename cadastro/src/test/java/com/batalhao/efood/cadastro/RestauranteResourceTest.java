package com.batalhao.efood.cadastro;

import javax.ws.rs.core.Response.Status;
import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import com.batalhao.efood.cadastro.dto.AtualizarRestauranteDTO;
import com.batalhao.efood.cadastro.model.Restaurante;
import com.batalhao.efood.cadastro.util.TokenUtils;
import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.http.Header;
import io.restassured.specification.RequestSpecification;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteResourceTest {

  private String token;

  @BeforeEach
  public void gereToken() throws Exception {
    token = TokenUtils.generateTokenString("/JWTProprietarioClaims.json", null);
  }

  private RequestSpecification given() {
    return RestAssured.given().contentType(ContentType.JSON)
        .header(new Header("Authorization", "Bearer " + token));
  }

  @Test
  @DataSet("restaurante-cenario-1.yml")
  public void testeBuscarRestaurantes() {
    String resultado = null;
    resultado = given().when().get("/restaurantes").then().statusCode(200).extract().asString();

    Approvals.verifyJson(resultado);
  }

  @Test
  @DataSet("restaurante-cenario-1.yml")
  public void testaAlterarRestaurante() {
    AtualizarRestauranteDTO dto = new AtualizarRestauranteDTO();
    dto.nomeFantasia = "Novo nome";
    Long parameterValue = 123L;
    given().with().pathParam("id", parameterValue).body(dto).when().put("/restaurantes/{id}").then()
        .statusCode(Status.NO_CONTENT.getStatusCode()).extract().asString();

    Restaurante findById = Restaurante.findById(parameterValue);
    // Assert.assertEquals(dto.nome, findById.nome);
    Assert.assertEquals(dto.nomeFantasia, findById.nome);
  }

}
