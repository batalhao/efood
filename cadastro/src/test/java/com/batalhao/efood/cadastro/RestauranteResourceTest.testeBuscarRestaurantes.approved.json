[
  {
    "id": 123,
    "nomeFantasia": "Novo Restaurante",
    "proprietario": "ID do Keyclock"
  },
  {
    "id": 234,
    "nomeFantasia": "Novo Restaurante 2",
    "proprietario": "ID do Keyclock 2"
  }
]